character.cpp

```diff
@@ -64,52 +65,17 @@ void Character::parseUnit(Character &C, std::string charSheetName)
 		throw std::runtime_error(error);
 	}
 
-	std::string line;
+	Parser parser;
+	std::map<std::string, std::string> attributes = parser.loadInput(charSheet);
 
-	while (!charSheet.eof())
-	{
-		std::getline(charSheet, line);
-
-		if (C.getName() == "")
-		{
-			std::string s = "name";
-			if (line.find(s) != std::string::npos) {
-				int end = line.rfind('"');
-				int start = end;
-				bool find = true;
-				while (find)
-				{
-					start--;
-					if (line[start] == '"')
-					{
-						find = false;
-					}
-				}
-				int length = end - start - 1;
-				C.name = line.substr(start + 1, length);
-			}
-		}
-
-		if (C.getHp() == 0)
-		{
-			std::string s = "hp";
-			if (line.find(s) != std::string::npos) {
-				int start = line.rfind(':');
-				int end = line.rfind(',');
-				int length = end - start - 2;
-				C.hp = std::stoi(line.substr(start + 2, length));
-			}
-		}
-
-		if (C.getDmg() == 0)
-		{
-			std::string s = "dmg";
-			if (line.find(s) != std::string::npos) {
-				int start = line.rfind(':');
-				int length = line.length() - start - 1;
-				C.dmg = std::stoi(line.substr(start + 2, length));
-			}
-		}
+	if(attributes.count("name")!=0 && attributes.count("hp")!=0 && attributes.count("dmg")!=0){
+		C.name = attributes["name"];
+		C.hp = std::stoi(attributes["hp"]);
+		C.dmg = std::stoi(attributes["dmg"]);
```

Comment:
Nem túl szép agyoncsapni egy létező példány változóit. Inkább konstruálni kellene egy új objektumot ezekkel a paraméterekkel és azt visszaadni.

Reply:
Javítva, most a statikus parseUnit függvény egy új Objektumot ad vissza.